Adds a technical report (#188)
dedis/dela: core/execution/benchmark/Report.md

```
Bench                Unikernel
┌─────────┐       ┌────────────┐
│         ├──────►│   Smart C. │
└─────────┘       └────────────┘
```

### Local TCP server

In this setup, the smart contract is run in go via a TCP connection.

```
Bench               go TCP server
┌──────────┐       ┌──────────┐
│          ├──────►│  Smart C.│
└──────────┘       └──────────┘
```

### Solidity in local

This setup uses the go-ethereum library to execute a solidity smart contract in
its VM from go.

```
Bench
┌───────────────┐
│               │
│         sol VM│
│     ┌─────────┤
│     │ Smart C.│
└─────┴─────────┘
 ```

### Solidity via TCP

This setup uses the go-ethereum library to execute a solidity smart contract via
a TCP server.

```
                    go TCP server
                   ┌────────────┐
Bench              │      Sol VM│
┌──────────┐       │  ┌─────────┤
│          ├──────►│  │ Smart C.│
└──────────┘       └──┴─────────┘
```

## Experiments

We perform two series of experiment:

1. **Increment**: in this series of experiment the smart contract increments
   the input it receives by 1.
2. **Simple crypto**: in this series of experiment the smart contract performs a

>    simple operation on ed25519 elliptic curves. The smart contract takes two
>    points (A,B) and a scalar (S) in argument. It then performs (A+B)*S.

Reviewer: ```suggestion
   simple operation on ed25519 elliptic curves. The smart contract takes a scalar (s) in argument. It then performs `s*G`, where `G` is the ed25519 base point.
```

## Results

|   [ns/op]    |Native|Unikernel |TCP   |Solidity|Solidity TCP|
|--------------|------|----------|------|--------|------------|
|Increment     |0.000 |0.470     |0.014 |0.001   |0.014       |

> |Simple crypto |0.005 |          |      |0.237   |            |

Reviewer 2: Effectively, we can consider that the "increment" implementation is as good as testing the networking overhead. I don't see a need to measure them separately for now.
